### ISSUE TITLE
cargo-install--path-. rebuilds everything from scratch, what's the point to first run cargo-build-release then?

### ISSUE THREAD

**Issue by laoshaw** [uutils/coreutils#4338]
I built coreutils and then install it per the instructions:

cargo build --release
cargo install --path .

I then noticed the second step just recompiles everything, do I still need the first cargo-build step at all?

**Comment by tertsdiepraam** [comment]
Those sections in the readme are indeed independent. I can see where the confusion comes from. Do you have a suggestion on how we could make that clearer?

**Comment by tertsdiepraam** [comment]
Yeah `--release` is the default for `cargo install`. I'm sure there are good reasons why `cargo install` always rebuilds, probably to ensure that you're installing the right thing, or to force rerun build scripts.

**Comment by oech3** [comment]
Still reproduce? No at reast for me. But adding `--locked` to everywhere might better to avoid unexpected update of crates and rebuild.
